site/src/pages/WorkspaceSchedulePage/WorkspaceSchedulePage.tsx

```diff
@@ -167,9 +116,66 @@ export const WorkspaceSchedulePage: React.FC = () => {
     username && workspaceName && scheduleSend({ type: "GET_WORKSPACE", username, workspaceName })
   }, [username, workspaceName, scheduleSend])
 
+  const getAutoStart = (workspace?: TypesGen.Workspace) =>
+    scheduleToAutoStart(workspace?.autostart_schedule)
+  const getAutoStop = (workspace?: TypesGen.Workspace) => ttlMsToAutoStop(workspace?.ttl_ms)
+
+  const [autoStart, setAutoStart] = useState(getAutoStart(workspace))
+  const [autoStop, setAutoStop] = useState(getAutoStop(workspace))
+
+  useEffect(() => {
+    setAutoStart(getAutoStart(workspace))
+    setAutoStop(getAutoStop(workspace))
+  }, [workspace])
```

Comment:
I don't mind this `useEffect`. If we want to have more control over it, you could change the dependency array to `[workspace?.autostart_schedule, workspace?.ttl_ms]`. Our linter will complain but I think this would be a decent case to override it.